packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -0,0 +1,2134 @@
+  ///
+  /// Menus commonly apply padding to the top and bottom of the menu surface,
+  /// which can cause a submenu's items to be vertically misaligned with their
+  /// parent menu items. To ensure a submenu's items align with their parent's
+  /// items, the [padding] applied to the menu surface is ignored when
+  /// calculating the position of the menu.
+  ///
+  /// Defaults to null.
+  final EdgeInsetsGeometry? padding;
+
+  /// The point on the menu surface that attaches to the anchor.
+  ///
+  /// Unlike [alignment] and [alignmentOffset], the [menuAlignment] will be
+  /// applied when the menu is opened with a `position` argument.
+  ///
+  /// Defaults to [AlignmentDirectional.topStart].
+  final AlignmentGeometry? menuAlignment;
+
+  // The semanticLabel argument is used by accessibility frameworks to announce
+  // the name of the menu.
+  final String? _semanticLabel;
+
+  /// {@template flutter.widgets.RawMenuAnchor.useRootOverlay}
+  /// Whether the menu panel should be rendered in the root [Overlay].
+  ///
+  /// When true, the menu is mounted in the root overlay. Rendering the menu in
+  /// the root overlay prevents the menu from being obscured by other widgets.
+  ///
+  /// When false, the menu is rendered in the nearest ancestor [Overlay].
+  ///
+  /// Submenus will always use the same overlay as their top-level ancestor, so
+  /// setting a [useRootOverlay] value on a submenu will have no effect.
+  /// {@endtemplate}
+  ///
+  /// Defaults to false on overlay menus.
+  final bool? useRootOverlay;
+
+  // The menu panel that is displayed when the menu is opened.
+  //
+  // The panel should lay out its menu children in a vertical list.
+  final Widget? _panel;
+  final TransitionBuilder? _nodeBuilder;
+  final RawMenuAnchorOverlayBuilder? _overlayBuilder;
+
+  /// The overlay builder used by the default [RawMenuAnchor] constructor.
+  ///
+  /// The [defaultOverlayBuilder] constructor builds a simple menu overlay. An
+  /// internal [FocusScope] is created to manage focus. Default keyboard
+  /// shortcuts include:
+  ///
+  /// - `ArrowUp` moves focus to the previous menu item.
+  /// - `ArrowDown` moves focus to the next menu item.
+  /// - `ArrowLeft` moves focus out of a submenu, thereby closing that submenu.
+  /// - `ArrowRight` opens and moves focus into a submenu.
+  /// - `Home` moves focus to the first menu item.
+  /// - `End` moves focus to the last menu item.
+  /// - `Escape` closes all menu layers.
+  ///
+  /// To customize the overlay, use the [RawMenuAnchor.overlayBuilder]
+  /// constructor.
+  Widget defaultOverlayBuilder(
+    BuildContext context,
+    RawMenuAnchorOverlayPosition position,
+  ) {
+    return _MenuOverlay(
+      position: position,
+      alignmentOffset: alignmentOffset,
+      alignment: alignment,
+      menuAlignment: menuAlignment,
+      consumeOutsideTaps: consumeOutsideTaps,
+      padding: padding,
+      semanticLabel: _semanticLabel,
+      child: _panel!,
+    );
```

Comment:
My preference would be to get rid of the `_MenuOverlay` class and put its build method here instead.

Reply:
I'll defer to @gspencergoog here because I based this off of the MenuAnchor. I think organizing the _MenuOverlay into a separate class makes it easier to understand the code, but I'm not as familiar with Flutter as Greg is.